src/transformers/pipelines/text_to_audio.py

```diff
@@ -0,0 +1,134 @@
+                SpeechT5HifiGan.from_pretrained(DEFAULT_VOCODER_ID).to(self.model.device)
+                if vocoder is None
+                else vocoder
+            )
+
+        self.sampling_rate = sampling_rate
+        if self.vocoder is not None:
+            self.sampling_rate = self.vocoder.config.sampling_rate
+
+        if self.sampling_rate is None:
+            # get sampling_rate from config and generation config
+            self.sampling_rate = None
+
+            config = self.model.config.to_dict()
+            gen_config = self.model.__dict__.get("generation_config", None)
+            if gen_config is not None:
+                config.update(gen_config.to_dict())
+
+            for sampling_rate_name in ["sample_rate", "sampling_rate"]:
+                sampling_rate = config.get(sampling_rate_name, None)
+                if sampling_rate is not None:
+                    self.sampling_rate = sampling_rate
+
+    def preprocess(self, text, **kwargs):
+        if isinstance(text, str):
+            text = [text]
+
+        output = self.tokenizer(text, **kwargs, return_tensors="pt")
+
+        return output
+
+    def _forward(self, model_inputs, **kwargs):
+        # we expect some kwargs to be additional tensors which need to be on the right device
+        kwargs = self._ensure_tensor_on_device(kwargs, device=self.device)
+
+        # call the generate by defaults or the forward method if the model cannot generate
+        output = self.forward_method(**model_inputs, **kwargs)
+
+        if self.vocoder is not None:
+            # in that case, the output is a spectrogram that needs to be converted into a waveform
+            output = self.vocoder(output)
+
+        return output
+
+    def __call__(
+        self,
+        input_texts: Union[str, List[str]],
+        **forward_params,
+    ):
+        """
+        Generates speech/audio from the inputs. See the [`TextToAudioPipeline`] documentation for more information.
+
+        Args:
+            input_texts (`str` or `List[str]`):
+                The text(s) to generate.
+            forward_params (*optional*):
+                Parameters passed to the model generation/forward method.
+
+        Return:
+            A `torch.Tensor` or a list of `torch.Tensor`: Each result comes as a `torch.Tensor` corresponding to the
+            generated audio.
+        """
+        return super().__call__(input_texts, **forward_params)
+
+    def _sanitize_parameters(
+        self,
+        preprocess_params=None,
+        forward_params=None,
+    ):
+        if preprocess_params is None:
+            preprocess_params = {}
+        if forward_params is None:
+            forward_params = {}
+        postprocess_params = {}
+
+        return preprocess_params, forward_params, postprocess_params
+
+    def postprocess(self, waveform):
+        output_dict = {}
+
+        output_dict["audio"] = waveform.detach().cpu().numpy().squeeze().astype(np.float32)
```

Comment:
Everything should be run in the `torch.no_grad()` context no? Either the forward pass or generation?

```suggestion
        output_dict["audio"] = waveform.cpu().numpy().squeeze().astype(np.float32)
```

Reply:
Are we happy with float32 numpy output as the output format? If so we can just do:
```suggestion
        output_dict["audio"] = waveform.cpu().float().numpy()
```
=> not sure what the extra `squeeze` was for here?

Reply:
For the extra squeeze, it is to return directly a 1D waveform to the user!

Reply:
Are we potentially loosing the "channel" dimension if we use a multi-channel encodec model to decode? Just thinking about if in the future we have model that outputs stereo audio instead of mono